dataplane/vrf.h

```diff
@@ -0,0 +1,91 @@
+#pragma once
+
+#include <array>
+#include <tuple>
+#include <vector>
+
+#include "common/blocks_allocator.h"
+#include "common/result.h"
+#include "lpm.h"
+#include "type.h"
+
+namespace dataplane::vrflpm
+{
+
+constexpr inline uint32_t lpmValueIdInvalid = dataplane::lpmValueIdInvalid;
+
+struct stats_t
+{
+	uint64_t extended_chunks_count;
+	uint64_t extended_chunks_size;
+};
+
+class VrfLpm4Linear
+{
+public:
+	constexpr static size_t size_of_chunk = 16;
+	constexpr static uint64_t extended_chunks_size_min = YANET_RIB_VRF_MAX_NUMBER;
+
+	using BlocksAllocator = common::allocator::BlocksAllocator<size_of_chunk>;
+
+	VrfLpm4Linear(BlocksAllocator& allocator);
+
+	eResult Insert(stats_t& stats, tVrfId vrfId, const uint32_t& ipAddress, const uint8_t& mask, const uint32_t& valueId);
+	eResult Remove(stats_t& stats, tVrfId vrfId, const uint32_t& ipAddress, const uint8_t& mask);
+	eResult Clear();
+	eResult CopyFrom(const VrfLpm4Linear& other, stats_t& stats);
+	void Swap(VrfLpm4Linear& other);
+	void Lookup(const uint32_t* ipAddresses, const tVrfId* vrfIds, uint32_t* valueIds, const unsigned int& count) const;
+	std::vector<std::tuple<tVrfId, uint32_t, uint8_t, uint32_t>> GetFullList() const;
+
+private:
+	struct OneBlock
+	{
+		VrfLpm4Linear::BlocksAllocator::Index next_block;
+		uint32_t ipAddress;
+		uint32_t valueId;
+		uint8_t mask;
+		uint8_t used;
+	};
+	static_assert(sizeof(OneBlock) <= size_of_chunk, "invalid size of OneBlock");
+
+	BlocksAllocator& allocator_;
+
```

Comment:
Do not use reference members. 
Whenever a reference is used it looks like value type (like `.` operator), but behaves like a pointer, that is can dangle, etc.
Plus now a lifetime of an `VrfLpm4Linear` object depends on a lifetime of an Allocator, but not in a safe way.

The usage of this class anyway looks like:
```cpp
//create buffer
std::unique_ptr<uint8_t[]> buffer = std::make_unique<uint8_t[]>(size_of_mem);
//create and init allocator
VrfLpm4Linear::BlocksAllocator allocator;
allocator.Init(buffer.get(), size_of_mem, YANET_RIB_VRF_MAX_NUMBER);
//create VrfLpm4Linear with allocator
VrfLpm4Linear vrf_linear(allocator);
```

So that allocator only gets used for this particular instance anyway.

Hence, you should define a move operations for `BlocksAllocator` (that's easy cause we already have a noexcept `Swap` method in it) 
```cpp
    BlocksAllocator(BlocksAllocator&& other) noexcept {
        Swap(other);
    }

    BlocksAllocator& operator=(BlocksAllocator&& other) noexcept {
        if (this != &other) {
            Swap(other);
        }
        return *this;
    }
```
and use `BlocksAllocator allocator_;` here with the constructor:

```cpp
VrfLpm4Linear::VrfLpm4Linear(BlocksAllocator&& allocator) :
        allocator_(std::move(allocator))
{
	Clear();
}
```